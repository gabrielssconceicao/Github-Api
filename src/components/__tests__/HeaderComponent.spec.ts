import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import HeaderComponentVue from '../HeaderComponent.vue';

describe('<HeaderComponent />', () => {
  it('should render header', () => {
    const wrapper = mount(HeaderComponentVue);
    expect(wrapper.findComponent('header').exists()).toBe(true);

    expect(wrapper.find('h1').text()).toBe('GitHub Finder');
    expect(wrapper.find('img[aria-label="GitHub Logo"]').exists()).toBe(true);
    expect(wrapper.find('input[name="search"]').exists()).toBe(true);
    expect(wrapper.find('button[type="submit"]').exists()).toBe(true);
  });

  it('should send search event', async () => {
    const wrapper = mount(HeaderComponentVue);

    const searchInput = wrapper.find('input[name="search"]');
    await searchInput.setValue('resolveTest');

    const form = wrapper.find('form');
    await form.trigger('submit');
    await wrapper.vm.$nextTick();

    expect(wrapper.emitted('submit')).toBeTruthy();
    expect(wrapper.emitted('submit')).toHaveLength(1);
  });

  it('should match snapshot', () => {
    const wrapper = mount(HeaderComponentVue);
    expect(wrapper.element).toMatchSnapshot();
  });
});
