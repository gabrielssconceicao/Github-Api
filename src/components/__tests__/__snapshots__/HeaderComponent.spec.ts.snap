// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<HeaderComponent /> > should match the snapshot 1`] = `
<header
  class="flex flex-col bg-header py-4 gap-3 md:flex-row md:justify-between md:px-4"
>
  <div
    class="flex justify-center items-center gap-4"
  >
    <img
      alt="GitHub Logo"
      aria-label="GitHub Logo"
      class="w-12"
      src="/src/assets/images/github-logo.svg"
    />
    <h1
      class="font-serif font-bold text-xl text-gray-300 tracking-wide"
    >
      GitHub Finder
    </h1>
  </div>
  <form
    autocomplete="off"
    class="flex items-stretch gap-3 px-3 py-2 md:flex-1 md:justify-end"
  >
    <input
      aria-label="Search"
      class="flex-1 rounded-3xl px-3 py-2 font-mono font-medium tracking-wider focus-visible:outline-none md:flex-1 md:max-w-[60%]"
      id="search"
      name="search"
      placeholder="Search GitHub users"
      type="text"
    />
    <button
      class="p-2 w-9 h-9 rounded-full bg-primary"
      type="submit"
    >
      <img
        alt="Search image"
        src="/src/assets/images/search.svg"
      />
    </button>
  </form>
</header>
`;
